Bug with the version 2.0.8 of spring boot (breakable change in the Document annotation)

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile (default-compile) on project spring-test-mongo: Compilation failure: Compilation failure: 
[ERROR] /spring-test-mongo/src/main/java/com/jupiter/tools/spring/test/mongo/internal/exportdata/scanner/ReflectionsDocumentScanner.java:[42,36] cannot find symbol
[ERROR]   symbol:   method value()
[ERROR]   location: variable annotation of type org.springframework.data.mongodb.core.mapping.Document
```

--- a/pom.xml
+++ b/pom.xml
@@ -5,7 +5,7 @@
 
     <groupId>com.antkorwin</groupId>
     <artifactId>spring-test-mongo</artifactId>
-    <version>0.12</version>
+    <version>0.13-SNAPSHOT</version>
     <packaging>jar</packaging>
 
     <name>spring-test-mongo</name>

--- a/pom.xml
+++ b/pom.xml
@@ -25,7 +25,7 @@
     </developers>
 
     <properties>
-        <spring-boot.version>[2.0.3.RELEASE, 2.1.3.RELEASE]</spring-boot.version>
+        <spring-boot.version>[2.0.3.RELEASE, 2.1.4.RELEASE]</spring-boot.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <java.version>1.8</java.version>

--- a/src/main/java/com/jupiter/tools/spring/test/mongo/internal/exportdata/scanner/ReflectionsDocumentScanner.java
+++ b/src/main/java/com/jupiter/tools/spring/test/mongo/internal/exportdata/scanner/ReflectionsDocumentScanner.java
@@ -1,6 +1,7 @@
 package com.jupiter.tools.spring.test.mongo.internal.exportdata.scanner;
 
 import org.reflections.Reflections;
+import org.springframework.core.annotation.AnnotationUtils;
 import org.springframework.data.mongodb.core.mapping.Document;
 
 import java.util.HashMap;
@@ -9,10 +10,10 @@ import java.util.Set;
 
 /**
  * Created on 03.01.2019.
- *
+ * <p>
  * Scans selected package on the MongoDB Documents and returns
  * a map with all founded collections with their class-types.
- *
+ * <p>
  * Finds all classes which annotated by the Document annotation
  * and takes collection names from this annotation or takes a class
  * name if the collection name doesn't set by the Document annotation.
@@ -38,24 +39,23 @@ public class ReflectionsDocumentScanner implements DocumentScanner {
         for (Class<?> doc : documents) {
 
             Document annotation = doc.getAnnotation(Document.class);
+            String value = (String) AnnotationUtils.getValue(annotation, "value");
 
-            if (notEmpty(annotation.value())) {
-                result.put(annotation.value(), doc);
-                continue;
+            if (empty(value)) {
+                value = annotation.collection();
             }
 
-            if (notEmpty(annotation.collection())) {
-                result.put(annotation.collection(), doc);
-                continue;
+            if(empty(value)){
+                value = doc.getSimpleName().toLowerCase();
             }
 
-            result.put(doc.getSimpleName().toLowerCase(), doc);
+            result.put(value, doc);
         }
 
         return result;
     }
 
-    private boolean notEmpty(String value) {
-        return !("".equals(value));
+    private boolean empty(String value) {
+        return value == null || "".equals(value);
     }
 }

--- a/src/test/java/com/jupiter/tools/spring/test/mongo/internal/exportdata/scanner/ReflectionsDocumentScannerTest.java
+++ b/src/test/java/com/jupiter/tools/spring/test/mongo/internal/exportdata/scanner/ReflectionsDocumentScannerTest.java
@@ -40,7 +40,9 @@ class ReflectionsDocumentScannerTest {
 
     }
 
-    @Document(value = "antkorwin-test-doc-second")
+    // since spring boot 2.1.x
+    // you can use value instead of the attribute `collection`:
+    @Document("antkorwin-test-doc-second")
     private class AntkorwinTestDocSecond {
 
     }

